dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,532 @@
+from __future__ import division
+
+import logging
+import math
+from warnings import warn
+
+import numpy as np
+from sklearn.metrics.scorer import check_scoring
+from sklearn.utils import check_random_state
+from tornado import gen
+
+from ._incremental import IncrementalSearchCV
+from ._successive_halving import SuccessiveHalvingSearchCV
+
+logger = logging.getLogger(__name__)
+
+
+def _get_hyperband_params(R, eta=3):
+    """
+    Parameters
+    ----------
+    R : int
+        The maximum number of iterations desired.
+
+    Returns
+    -------
+    brackets : Dict[int, Tuple[int, int]]
+        A dictionary of the form {bracket_id: (n_models, n_initial_iter)}
+
+    Notes
+    -----
+    The bracket index is a measure of how strong that n,r combination
+    adapts to prior input. i.e., a bracket ID of 0 means "doesn't adapt
+    at all" and bracket index of 5 means "adapts pretty strongly"
+
+    """
+    s_max = math.floor(math.log(R, eta))
+    B = (s_max + 1) * R
+
+    brackets = list(reversed(range(int(s_max + 1))))
+    N = [int(math.ceil(B / R * eta ** s / (s + 1))) for s in brackets]
+    R = [int(R * eta ** -s) for s in brackets]
+    return {b: (n, r) for b, n, r in zip(brackets, N, R)}
+
+
+class HyperbandSearchCV(IncrementalSearchCV):
+    """Find the best parameters for a particular model with an adaptive
+    cross-validation algorithm.
+
+    Hyperband will find close to the best possible parameters with
+    the given computational budget [*]_ by spending more time training
+    high-performing estimators [1]_. This means that Hyperband stops training
+    estimators that perform poorly.
+
+    Hyperband only requires computational budget as input, and
+    does not require a trade-off between "evaluate many parameters
+    for a short time" and "train a few parameters for a long time"
+    like RandomizedSearchCV.
+
+    .. [*] After :math:`N` ``partial_fit`` calls the estimator Hyperband
+       produces will be close to the best possible estimator that :math:`N`
+       ``partial_fit`` calls could ever produce with high probability (where
+       "close" means "within log terms").
+
+    Parameters
+    ----------
+    estimator : object
```

Comment:
Use the description from IncrementalSearchCV? I find that a bit clearer than this (and successive halving's)